content/kb/tutorials/deploy/kubernetes.md

```diff
@@ -0,0 +1,288 @@
+
+Here's an example `Dockerfile` that you can add to the root of your directory.
+
+```dockerfile
+FROM python:3.7-slim
+
+RUN groupadd --gid 1000 appuser \
+    && useradd --uid 1000 --gid 1000 -ms /bin/bash appuser
+
+RUN pip3 install --no-cache-dir --upgrade \
+    pip \
+    virtualenv
+
+RUN apt-get update && apt-get install -y \
+    build-essential \
+    software-properties-common \
+    git
+
+USER appuser
+WORKDIR /home/appuser
+
+RUN git clone https://github.com/streamlit/streamlit-example.git app
+
+ENV VIRTUAL_ENV=/home/appuser/venv
+RUN virtualenv ${VIRTUAL_ENV}
+RUN . ${VIRTUAL_ENV}/bin/activate && pip install -r app/requirements.txt
+
+EXPOSE 8501
+
+COPY run.sh /home/appuser
+ENTRYPOINT ["./run.sh"]
+```
+
+### Build a Docker image
+
+Put the above files (`run.sh` and `Dockerfile`) in the same folder and build the docker image:
+
+```docker
+docker build --platform linux/amd64 -t gcr.io/$GCP_PROJECT_ID/k8s-streamlit:test .
+```
+
+<Important>
+
+Replace `$GCP_PROJECT_ID` in the above command with the name of your Google Cloud project.
+
+</Important>
+
+### Upload the Docker image to a container registry
+
+The next step is to upload the Docker image to a container registry. In this example, we will use the [Google Container Registry (GCR)](https://cloud.google.com/container-registry). Start by enabling the Container Registry API. Sign in to Google Cloud and navigate to your project’s **Container Registry** and click **Enable**.
+
+We can now build the Docker image from the previous step and push it to our project’s GCR. Be sure to replace `$GCP_PROJECT_ID` in the docker push command with the name of your project:
+
+```bash
+gcloud auth configure-docker
+docker push gcr.io/s4t-head/k8s-streamlit:test
+```
+
+## Create a Kubernetes deployment
+
+For this step you will need a:
+
+- Running Kubernetes service
+- Custom domain for which you can generate a TLS certificate
+- DNS service where you can configure your custom domain to point to the application IP
+
+As the image was uploaded to the container registry in the previous step, we can run it in Kubernetes using the below configurations.
+
+### Install and run Kubernetes
+
+Make sure your Kubernetes client is installed and running on your machine ([guide](https://kubernetes.io/docs/tasks/tools/#kubectl)).
+
+### Configure a Google OAuth Client and oauth2-proxy
+
+For configuring Google OAuth Client, please see [this](https://oauth2-proxy.github.io/oauth2-proxy/docs/configuration/oauth_provider#google-auth-provider) document. Configure oauth2-proxy to use the desired oauth provider: [guide](https://oauth2-proxy.github.io/oauth2-proxy/docs/configuration/oauth_provider) and update the oath2-proxy config in the config map.
```

Comment:
Can you use meaningful [link text](https://developers.google.com/style/link-text)?